app/Http/Controllers/AuthController.php

```diff
@@ -3,11 +3,198 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use GuzzleHttp\Psr7;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Client;
+use Illuminate\Support\Facades\Session;
+
 
 class AuthController extends Controller
 {
+    protected $baseURL;
+
+    public function __construct()
+    {
+        $this->baseURL = env('APP_URL');
+    }
+
+    public function index()
+    {
+        //Session::put('name', "isfhan Ahmed");
+        //return '<h3>Welcome to Findologic controller</h3> <br> <a href="auth/hello-world">click here</a>';
+        return "<h3>Welcome to Findologic controller</h3>";
+    }
+
+    public function getAppClientId()
+    {
+        return env('DEV_CLIENT_ID');
+    }
+
+    public function getAppSecret(Request $request)
+    {
+        return env('DEV_CLIENT_SECRET');
+    }
+
+    public function getAccessToken(Request $request)
+    {
+        return Session::get('access_token');
+    }
+
+
+    public function getStoreHash(Request $request)
+    {
+        return Session::get('store_hash');
+    }
+
+
     public function helloWorld(Request $request)
     {
+        //return 'Hello World!'.Session::get('name')."";
         return 'Hello World!';
     }
+
+    public function insatll(Request $request)
+    {
+        //return 'insatll route hit';
+
+        // Make sure all required query params have been passed
+        if (!$request->has('code') || !$request->has('scope') || !$request->has('context')) {
+            $errorMessage = "Not enough information was passed to install this app.";
+            echo '<h4>An issue has occurred:</h4> <p>' . $errorMessage . '</p> <a href="'.$this->baseURL.'">Go back to home</a>';
+        }
+
+        try {
+            $client = new Client();
+            $result = $client->request('POST', 'https://login.bigcommerce.com/oauth2/token', [
+                'json' => [
+                    'client_id' => $this->getAppClientId(),
+                    'client_secret' => $this->getAppSecret($request),
+                    'redirect_uri' => $this->baseURL . '/auth/install',
+                    'grant_type' => 'authorization_code',
+                    'code' => $request->input('code'),
+                    'scope' => $request->input('scope'),
+                    'context' => $request->input('context'),
+                ]
+            ]);
+
+            $statusCode = $result->getStatusCode();
+            // Inside data you have access_token,context,userId,userEmail
+            $data = json_decode($result->getBody(), true);
+
+            if ($statusCode == 200) {
+                Session::put('store_hash', $data['context']);
+                Session::put('access_token', $data['access_token']);
```

Comment:
The access_token must be stored in the db for later use (data export). I would suggest to modify the migration of the stores table and set the same four fields as we get here.